src/utils/file.ts

```diff
@@ -1,7 +1,16 @@
 import path from 'path'
 import fs from 'fs'
+import { TestInfo } from '@playwright/test';
 
 export async function readFile(filename: string) {
   const filePath = path.join(__dirname, '../config/templates', filename)
   return fs.promises.readFile(filePath, { encoding: 'utf-8' })
 }
+
+export async function reportFileExists(testInfo: TestInfo, reportFileName: string) {
+  const reportFilePath = testInfo.attachments.find(attachment => attachment.name === reportFileName)?.path
+  if (!reportFilePath) return false
+  const filePath = path.join(__dirname, '../../', reportFilePath)
+  try { fs.promises.access(filePath) } catch { return false }
+  return true
```

Comment:
Seems like we can just test for the presence in the test info attachments right?  These three lines shouldn't be needed.